### PR TITLE
Ilr ordination

### PR DESCRIPTION
**Pull request by mortonjt**
This is addressing #70 , outputting an ordination results object containing ILR coordinates.

CC @ElDeveloper , @fedarko

### CODE REVIEW

**Comment by mortonjt** [comment]
oof, I'm having an embarrassingly hard time creating this new metadata object.

Below is the latest error.
```
Plugin error from gneiss:

  No transformation from <class 'pandas.core.frame.DataFrame'> to <class 'qiime2.plugin.model.directory_format.CladeMetadataDirectoryFormat'>

Debug info has been saved to /tmp/qiime2-q2cli-err-yb6wyqgd.log
```
I probably made a dumb typo somewhere - thoughts @ebolyen?

File: q2_gneiss/composition/_method.py [hunk 1]
```diff
@@ -28,3 +31,98 @@ def ilr_phylogenetic(table: pd.DataFrame, tree: skbio.TreeNode,
...
+                                clades: list = None) -> (
+                                    OrdinationResults,
+                                    skbio.TreeNode, pd.DataFrame
+                                ):
+    t = tree.copy()
+    t.bifurcate()
+    _table, _tree = match_tips(table, t)
+    _tree = rename_internal_nodes(_tree)
+    if not clades:
+        clades = var.index[:top_k_var]
```
Comment (thermokarst) [inline q2_gneiss/composition/_method.py]: `var` is undefined here - is this a typo? Should it be `table`?

**Review by thermokarst — CHANGES_REQUESTED** [review]

**Comment by thermokarst** [comment]
Hey there @mortonjt, sorry for the slow reply!

> hard time creating this new metadata object.

Looks like we don't have a transformer defined for this in q2-types: https://github.com/qiime2/q2-types/blob/af9e2be5743f98db5ed87c81c23ac13756b8186e/q2_types/feature_data/_transformer.py#L400-L404

Let me know if you need a hand! Sorry for the last-minute notice, but the release deadline for PR merges is Nov 18th.

**Comment by mortonjt** [comment]
Hi Matt, thank you for taking a look at this!

I don't think I'll be able to push this in time unfortunately (I'm against
a very tight manuscript deadline) -- but I will definitely factor in these
comments for the next release.

File: q2_gneiss/composition/_method.py (hunk 1 above)
Comment (mortonjt) [reply q2_gneiss/composition/_method.py]: Hi @thermokarst thank you for catching that.  There was a bit of shuffling that knocked the lines out of order.
That is now fixed and tested.

**Comment by mortonjt** [comment]
Hi @thermokarst thank you for the review - I've addressed the changed.  Let me know if you need anything else.

**Comment by mortonjt** [comment]
Hi @thermokarst , can we merge in this pull request? This is becoming a bottleneck in some of the Empress analyses - we could really use this sort of functionality :)

**Comment by thermokarst** [comment]
Is empress using a QIIME 2 Staging env? If not, even if we merge today, it won't be out until end of February.

**Comment by mortonjt** [comment]
Got it. No empress isn’t using the staged version. Can we hit the Feb
deadline?

**Review by thermokarst — APPROVED** [review]

**Comment by mortonjt** [comment]
Thank you!

**Comment by thermokarst** [comment]
@mortonjt   - do you think you can add a short entry for this change into the release changelog?

https://forum.qiime2.org/t/preview-qiime-2-2021-2-development-changelog/17815

**Comment by mortonjt** [comment]
absolutely! adding in information in a bit.

**Pull request merged**